pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -445,6 +456,90 @@ func (ctx *context) handleInboundRequest(request *Request, options *options,
 	}, connRec, nil
 }
 
+func (ctx *context) prepareResponse(request *Request, responseDidDoc *did.Doc,
+	responseConnection *Connection) (*Response, error) {
+	// prepare the response
+	response := &Response{
+		Type: ResponseMsgType,
+		ID:   uuid.New().String(),
+		Thread: &decorator.Thread{
+			ID: request.ID,
+		},
+	}
+
+	if request.Thread != nil {
+		response.Thread.PID = request.Thread.PID
+	}
+
+	if ctx.doACAPyInterop {
+		return ctx.prepareResponseInterop(request, response, responseDidDoc)
+	}
+
+	// prepare connection signature
+	encodedConnectionSignature, err := ctx.prepareConnectionSignature(responseConnection, request.Thread.PID)
+	if err != nil {
+		return nil, fmt.Errorf("connection signature: %w", err)
+	}
+
+	response.ConnectionSignature = encodedConnectionSignature
+
+	return response, nil
+}
+
+func (ctx *context) prepareResponseInterop(request *Request, response *Response,
```

Comment:
Missing comment on what is the AcaPy interop difference and link to open issue in AcaPy.

Reply:
renamed method to be more clear, and added an explanatory comment where it's invoked - this isn't an interop fix, it's an AIP2.0 fix, and making this behaviour default (instead of limited to interop testing with aca-py) will be my next PR.